packages/ui/src/components/primitives/message/attachments.tsx

```diff
@@ -41,26 +42,53 @@ const SingularImageAttachment = async (
 	);
 };
 
-export const MessageAttachments = (
-	props: Pick<MessageProps, 'message' | 'loadingStyle'> & {
-		limit?: number;
-	},
+const ImageAttachments = (
+	props: Pick<MessageProps, 'message' | 'loadingStyle'>,
 ) => {
 	const { message } = props;
 	const imageFileRegex = new RegExp('(.*/)*.+.(png|jpg|gif|bmp|jpeg|webp)$');
-	// TODO: Do not mutate here, ugly
+
 	message.attachments = message.attachments.filter((attachment) =>
 		imageFileRegex.test(attachment.filename.toLowerCase()),
 	);
 	if (message.attachments.length === 0) return null;
-	if (props.limit)
-		message.attachments = message.attachments.slice(0, props.limit);
 
-	// TODO: Rename this and such, whole file needs a revisit but I'm on vacation 🌴
 	return (
 		<SingularImageAttachment
 			message={message}
 			loadingStyle={props.loadingStyle}
 		/>
 	);
 };
+
+export const MessageAttachments = (
+	props: Pick<MessageProps, 'message' | 'loadingStyle'> & {
+		limit?: number;
+	},
+) => {
+	const { message } = props;
+
+	if (message.attachments.length === 0) return null;
+	if (props.limit)
+		message.attachments = message.attachments.slice(0, props.limit);
+	const imageFileRegex = new RegExp('(.*/)*.+.(png|jpg|gif|bmp|jpeg|webp)$');
+	const imageAttachments = message.attachments.filter((attachment) =>
+		imageFileRegex.test(attachment.filename.toLowerCase()),
+	);
+	if (imageAttachments.length > 0) {
+		return <ImageAttachments message={message} />;
```

Comment:
Make sure multi attachment types work here, image + files should
Show both

Reply:
Fixed! 😄🚀